controllers/scopeinstance_controller.go

```diff
@@ -369,9 +378,80 @@ func (r *ScopeInstanceReconciler) mapToScopeInstance(obj client.Object) (request
 		request := reconcile.Request{
 			NamespacedName: types.NamespacedName{Namespace: si.GetNamespace(), Name: si.GetName()},
 		}
-
 		requests = append(requests, request)
 	}
 
 	return
 }
+
+// getClusterRoleBindingForClusterRoleTemplate will create a ClusterRoleBinding from a ClusterRoleTemplate, ScopeInstance, and ScopeTemplate
+func (r *ScopeInstanceReconciler) getClusterRoleBinding(cr *operatorsv1.ClusterRoleTemplate, in *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate) *rbacv1.ClusterRoleBinding {
+	crb := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: cr.GenerateName + "-",
+			Labels: map[string]string{
+				scopeInstanceUIDKey:           string(in.GetUID()),
+				referenceHashKey:              hashScopeInstanceAndTemplate(in, st),
+				clusterRoleBindingGenerateKey: cr.GenerateName,
+			},
+		},
+		Subjects: cr.Subjects,
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     cr.GenerateName,
+			APIGroup: rbacv1.GroupName,
+		},
+	}
+
+	err := ctrl.SetControllerReference(in, crb, r.Scheme)
+	if err != nil {
+		log.Log.Error(err, "setting controller reference for ClusterRoleBinding")
+	}
+	return crb
+}
+
+// getRoleBindingForClusterRoleTemplate will create a ClusterRoleBinding from a ClusterRoleTemplate
+func (r *ScopeInstanceReconciler) getRoleBinding(cr *operatorsv1.ClusterRoleTemplate, in *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate, namespace string) *rbacv1.RoleBinding {
+	rb := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: cr.GenerateName + "-",
+			Namespace:    namespace,
+			Labels: map[string]string{
+				scopeInstanceUIDKey:           string(in.GetUID()),
+				referenceHashKey:              hashScopeInstanceAndTemplate(in, st),
+				clusterRoleBindingGenerateKey: cr.GenerateName,
+			},
+		},
+		Subjects: cr.Subjects,
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     cr.GenerateName,
+			APIGroup: rbacv1.GroupName,
+		},
+	}
+
+	err := ctrl.SetControllerReference(in, rb, r.Scheme)
+	if err != nil {
+		log.Log.Error(err, "setting controller reference for ClusterRoleBinding")
+	}
+	return rb
+}
+
+// referenchHash is used to store a ScopeInstance.Spec
+// and ScopeTemplate.Spec. This object is used for getting
+// the combined hash of both specs.
+type referenceHash struct {
+	ScopeInstanceSpec *operatorsv1.ScopeInstanceSpec
+	ScopeTemplateSpec *operatorsv1.ScopeTemplateSpec
+}
+
+// hashScopeInstanceAndTemplate will take in a ScopeInstance and ScopeTemplate and return
+// a combined hash of the ScopeInstance.Spec and ScopeTemplate.Spec fields
+func hashScopeInstanceAndTemplate(si *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate) string {
+	hashObj := &referenceHash{
+		ScopeInstanceSpec: &si.Spec,
+		ScopeTemplateSpec: &st.Spec,
+	}
+
+	return util.HashObject(hashObj)
+}
```

Comment:
A couple general questions on this:
- Is this okay in the controller code or would it be best to move this into the `utils/` pkg? 
    - I only foresee us using this in the `scopeinstance_controller` which is why I left it here, but I am open to moving it to the `utils/` pkg
- Since the `referenceHash` struct is only used during the `hashScopeInstanceAndTemplate` function would it be better to use it as an anonymous struct within the `hashScopeInstanceAndTemplate` function? Something like:
```go
func hashScopeInstanceAndTemplate(si *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate) string {
	hashObj := new struct{
	       ScopeInstanceSpec *operatorsv1.ScopeInstanceSpec
	       ScopeTemplateSpec *operatorsv1.ScopeTemplateSpec
	}{
		ScopeInstanceSpec: &si.Spec,
		ScopeTemplateSpec: &st.Spec,
	}

	return util.HashObject(hashObj)
}
```